lib/src/css_printer.dart

```diff
@@ -7,35 +7,89 @@ part of '../visitor.dart';
 /// Visitor that produces a formatted string representation of the CSS tree.
 class CssPrinter extends Visitor {
   StringBuffer _buff = StringBuffer();
-  bool prettyPrint = true;
+  bool _prettyPrint = true;
   bool _isInKeyframes = false;
+  int _indent = 0;
+  bool _startOfLine = true;
 
   /// Walk the [tree] Stylesheet. [pretty] if true emits line breaks, extra
   /// spaces, friendly property values, etc., if false emits compacted output.
   @override
   void visitTree(StyleSheet tree, {bool pretty = false}) {
-    prettyPrint = pretty;
+    _prettyPrint = pretty;
     _buff = StringBuffer();
+    _indent = 0;
+    _startOfLine = true;
+
     visitStyleSheet(tree);
   }
 
   /// Appends [str] to the output buffer.
   void emit(String str) {
-    _buff.write(str);
+    if (_prettyPrint) {
+      if (_startOfLine) {
+        _startOfLine = false;
+        _buff.write(' ' * _indent);
+      }
+      _buff.write(str);
+    } else {
+      _buff.write(str);
+    }
+  }
+
+  void _emitLBrace() {
+    if (_prettyPrint) {
+      _buff.write('{\n');
+      _startOfLine = true;
+      _indent += 2;
```

Comment:
nit: Since the else branch is a subset of the if branch, we can just extract the common parts outside the if-else:

```dart
_indent += 2;
_buff.write('{');

if (_prettyPrint) {
  _buff.writeln();
  _startOfLine = true;
}
```

Reply:
👍 - done